### PR TITLE
feat(user feedback): input form

### PR DESCRIPTION
**Pull request by armcknight**
for https://github.com/getsentry/sentry-cocoa/issues/4187?issue=getsentry%7Csentry-cocoa%7C4271; #skip-changelog

this is not feature complete or full polish for the form, we just want to shift towards iterating on end-to-end functionality, and then circle back to completing the implementation and UI polish.

functionality still needed after this PR, so should not be tested here:
- error states if required fields are not filled
- adding/removing screenshots
- automatically filling sentry user name/email

known issues:
- custom outline colors break text field padding
- logo doesn't scale and is sometimes clipped with certain font sizes

continued in https://github.com/getsentry/sentry-cocoa/pull/4544

### CODE REVIEW

**Review by brustolin — APPROVED** [review]
LGTM.

While reviewing, I considered that it might be worth exploring making the SDK options immutable, and since this is a new feature, we could start here. This could better align with supporting strict concurrency in Swift in the future. 
Instead of using a callback to configure the form and widget, we could expect the settings object to be directly assigned within the options.

One potential approach could be using a builder pattern, although it’s not my preferred solution.

**Review by philipphofmann — COMMENTED** [review]
Just out of curiosity: what drove you not to use SwiftUI for this, @armcknight?

**Comment by armcknight** [comment]
> Just out of curiosity: what drove you not to use SwiftUI for this, @armcknight?

@philipphofmann, SwiftUI in my opinion is still too immature. We could easily wind up in a situation where we can't do something we want with the UI, whereas with UIKit we have full control, and the developer experience is still subpar.

File: Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackForm.swift [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        delegate?.confirmed()
+    }
+    
+    func cancelButtonTapped() {
+        delegate?.cancelled()
+    }
+    
+    // MARK: Layout
+    
+    let formElementHeight: CGFloat = 40
```
[the remark this answers is not in the document]
Comment (armcknight) [reply Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackForm.swift]: I prefer knowing what something does higher up, and then putting the finer details lower down, that can be navigated to with ⌘ ⤓

**Comment by philipphofmann** [comment]
> @philipphofmann, SwiftUI in my opinion is still too immature. We could easily wind up in a situation where we can't do something we want with the UI, whereas with UIKit we have full control, and the developer experience is still subpar.

That's what I assumed. I heard that frequently.

**Pull request merged**